firedrake/checkpointing.py

```diff
@@ -871,6 +956,10 @@ def _save_function_topology(self, tf, idx=None):
                     topology_dm.globalVectorView(self.viewer, dm, vec)
                     topology_dm.setName(base_tmesh_name)
         if idx is not None:
+            has_indices_attr = self.has_attr(path, PREFIX_TIMESTEPPING + "_indices")
+            old_indices = self.get_attr(path, PREFIX_TIMESTEPPING + "_indices") if has_indices_attr else []
+            indices = np.concatenate((old_indices, [idx]))
+            self.set_attr(path, PREFIX_TIMESTEPPING + "_indices", indices)
```

Comment:
@ksagiyam Please check if this makes sense. I think all the checks above in timestepping should make this harmless?

Reply:
Ah right! They are now after exiting time-stepping mode.